packages/jest/src/index.js

```diff
@@ -47,19 +47,23 @@ module.exports = (options = {}) => (neutrino) => {
     const { extensions, source, tests, root, debug } = neutrino.options;
     const modulesConfig = neutrino.config.resolve.modules.values();
     const aliases = neutrino.config.resolve.alias.entries() || {};
+    const moduleFileExtensions = neutrino.config.resolve.extensions
+      .values()
+      // Jest does not yet support ES6 modules, see:
+      // https://github.com/facebook/jest/issues/4842
+      .filter((ext) => ext !== '.mjs')
+      .map((extension) => extension.replace('.', ''));
+    const extensionsOptions = moduleFileExtensions.length
+      ? { moduleFileExtensions }
+      : undefined;
 
     return merge(
       {
         rootDir: root,
+        ...extensionsOptions,
         moduleDirectories: modulesConfig.length
           ? modulesConfig
           : ['node_modules'],
```

Comment:
To simplify this slightly, we can use the spread operator with a conditional statement, like so:

```suggestion

    return merge(
      {
        rootDir: root,
        moduleDirectories: modulesConfig.length
          ? modulesConfig
          : ['node_modules'],
        ...(moduleFileExtensions.length && { moduleFileExtensions }),
```

This is similar to what we do in a few other places, eg:
https://github.com/neutrinojs/neutrino/blob/b740f67a546a8efbb0e59a890ab4a1ab4b3ae9a2/packages/web/index.js#L144